src/Pagination/Pagination.js

```diff
@@ -1,70 +1,110 @@
 
 const Page = styled.a`
-  position: relative;
-  float: left;
-  padding: 7px 12px;
-  margin-left: -1px;
-  font-size: ${get('pagination.fontSize')};
+  display: inline-block;
+  min-width: 32px;
+  padding: 5px 10px;
   font-style: normal;
-  font-weight: ${get('pagination.fontWeight')};
+  line-height: 20px;
   color: ${get('pagination.colors.normal.fg')};
+  text-align: center;
   white-space: nowrap;
   vertical-align: middle;
   cursor: pointer;
   user-select: none;
-  background: ${get('pagination.colors.normal.bg')};
-  border: ${get('borderWidths.1')} solid ${get('pagination.colors.normal.border')};
   text-decoration: none;
 
-  &:first-child {
-    margin-left: 0;
-    border-top-left-radius: ${get('pagination.borderRadius')};
-    border-bottom-left-radius: ${get('pagination.borderRadius')};
-  }
+  margin-right: ${get('pagination.spaceBetween')};
 
   &:last-child {
-    border-top-right-radius: ${get('pagination.borderRadius')};
-    border-bottom-right-radius: ${get('pagination.borderRadius')};
+    margin-right: 0;
+  }
+
+  border: ${get('borderWidths.1')} solid transparent;
+  border-radius: ${get('pagination.borderRadius')};
+  transition: border-color 0.2s cubic-bezier(0.3, 0, 0.5, 1);
+
+  &:hover,
+  &:focus {
+    text-decoration: none;
+    border-color: ${get('pagination.colors.hover.border')};
+    outline: 0;
+    transition-duration: 0.1s;
+  }
+
+  &:active {
+    border-color: ${get('pagination.colors.active.border')};
   }
 
-  ${props =>
-    !props.selected &&
-    !props.disabled &&
-    css`
-      &:hover,
-      &:focus {
-        z-index: 2;
-        color: ${get('pagination.colors.hover.fg')};
-        background-color: ${get('pagination.colors.hover.bg')};
-        border-color: ${get('pagination.colors.hover.border')};
-      }
-    `}
-
-  ${props =>
-    props.selected &&
-    css`
-      z-index: 3;
-      color: ${get('pagination.colors.selected.fg')};
-      background-color: ${get('pagination.colors.selected.bg')};
-      border-color: ${get('pagination.colors.selected.border')};
-    `}
-
-  ${props =>
-    props.disabled &&
-    css`
-      color: ${get('pagination.colors.disabled.fg')};
-      cursor: default;
-      background-color: ${get('pagination.colors.disabled.bg')};
-      border-color: ${get('pagination.colors.disabled.border')};
-    `}
+  &[rel='prev'],
+  &[rel='next'] {
+    color: ${get('pagination.colors.nextPrevious.fg')};
+  }
+
+  &[aria-current],
+  &[aria-current]:hover {
+    color: ${get('pagination.colors.selected.fg')};
+    background-color: ${get('pagination.colors.selected.bg')};
+    border-color: ${get('pagination.colors.selected.border')};
+  }
+
+  &[aria-disabled],
+  &[aria-disabled]:hover {
+    color: ${get('pagination.colors.disabled.fg')}; // check
+    cursor: default;
+    border-color: ${get('pagination.colors.disabled.border')};
+  }
+
+  @supports (clip-path: polygon(50% 0, 100% 50%, 50% 100%)) {
```

Comment:
fancy ✨

Reply:
@simurai is a super fancy guy

Reply:
btw. this is kinda a "hack" :sweat_smile: because on dotcom there are different implementations of the pagination component and we first have to refactor that. Eventually this `clip-path` version could be replaced with a normal chevron octicon.